app/src/main/java/team7/voluntime/Fragments/Charities/CharityViewPreviousEventsFragment.java

```diff
@@ -0,0 +1,150 @@
+import team7.voluntime.Domains.Event;
+import team7.voluntime.R;
+import team7.voluntime.Utilities.EventListAdapter;
+import team7.voluntime.Utilities.Utilities;
+
+
+public class CharityViewPreviousEventsFragment extends Fragment {
+    private final static String TAG = "CharityViewEvents";
+    FirebaseUser mUser;
+    //ListView listOfUpcomingEventsLV;
+    ListView listOfPreviousEventsLV;
+    private FirebaseDatabase database;
+    private DatabaseReference charityReference;
+    private DatabaseReference eventReference;
+    private Charity charity;
+
+    public CharityViewPreviousEventsFragment() {
+        // Required empty public constructor
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        getActivity().setTitle("View Previous Events");
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        View v = inflater.inflate(R.layout.fragment_charity_view_previous_events, container, false);
+        ButterKnife.bind(this, v);
+        mUser = FirebaseAuth.getInstance().getCurrentUser();
+        database = FirebaseDatabase.getInstance();
+        charityReference = Utilities.getCharityReference(database, mUser.getUid());
+        eventReference = Utilities.getEventsReference(database);
+
+        listOfPreviousEventsLV = v.findViewById(R.id.listOfPreviousEventsLV);
+        final ArrayList<Event> previousEventList = new ArrayList<>();
+
+        final HashMap<String, String> previousEvents = new HashMap<>();
+
+        EventListAdapter<CharityViewPreviousEventsFragment> previousEventListAdapter = new EventListAdapter<>(getActivity(), R.layout.adapter_view_event_layout, previousEventList, CharityViewPreviousEventsFragment.this);
+
+        listOfPreviousEventsLV.setAdapter(previousEventListAdapter);
+
+        // Attach a listener to read the data at our posts reference
+        charityReference.child("Profile").addListenerForSingleValueEvent(new ValueEventListener() {
+            @Override
+            public void onDataChange(DataSnapshot dataSnapshot) {
+                charity = dataSnapshot.getValue(Charity.class);
+                charity.setId(mUser.getUid());
+                Log.d(TAG, charity.toString());
+            }
+
+            @Override
+            public void onCancelled(DatabaseError databaseError) {
+                Log.e(TAG, "The read failed: " + databaseError.getCode());
+            }
+        });
+
+        charityReference.child("Events").addListenerForSingleValueEvent(new ValueEventListener() {
+            @Override
+            public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
+                for (DataSnapshot child : dataSnapshot.getChildren()) {
+                    if (child.exists()) {
+                        if (child.getValue() != null) {
```

Comment:
Rather than nesting two if conditions do `if (child.exists() && child.getValue() != null)`

Reply:
Fixed in latest PR